pkg/metallb/frr_test.go

```diff
@@ -0,0 +1,510 @@
+package metallb
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/schemes/metallb/frrtypes"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	frrTestSchemes                = []clients.SchemeAttacher{frrtypes.AddToScheme}
+	defaultFrrConfigurationName   = "frrconfiguration"
+	defaultFrrConfigurationNsName = "test-namespace"
+	apiClient                     *clients.Settings
+)
+
+func TestNewFrrConfigurationBuilder(t *testing.T) {
+	generateFrrConfiguration := NewFrrConfigurationBuilder
+
+	testCases := []struct {
```

Comment:
needs to have a test case for when apiClient is nil, [mco/mcp_test.go](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/mco/mcp_test.go#L58-L66) has an example

Reply:
added

Reply:
@gkopels this always uses a client in the test so it doesn't actually test when apiClient is nil, see the mcp example for how to test when it's nil